incuna_auth/middleware/login_required.py

```diff
@@ -38,40 +38,33 @@ class LoginRequiredMiddleware:
     Will default to protecting everything if LOGIN_PROTECTED_URLS is not in
     settings.
 
-
     Original code from:
     http://onecreativeblog.com/post/59051248/django-login-required-middleware
 
     This version has been modified to allow us to define areas of the site to
     password protect instead of protecting everything under /.
-    '''
-    def process_request(self, request):
-        assert hasattr(request, 'user'), (
-            "The 'LoginRequiredMiddleware' "
-            + "requires authentication middleware to be installed. Edit your "
-            + "'MIDDLEWARE_CLASSES' setting to insert "
-            + "'django.contrib.auth.middlware.AuthenticationMiddleware'. "
-            + "If that doesn't work, ensure your 'TEMPLATE_CONTEXT_PROCESSORS' "
-            + "setting includes 'django.core.context_processors.auth'."
-        )
-
-        # Jump over this middleware if not a protected url.
-        path = request.path_info.lstrip('/')
-        path_is_exempt = any(m.match(path) for m in EXEMPT_URLS)
-        path_is_protected = any(m.match(path) for m in PROTECTED_URLS)
-        if path_is_exempt or not path_is_protected:
-            return
-
-        # Jump over this middleware if user logged in.
-        if request.user.is_authenticated():
-            return
-
-        # Raise a 403 for POST/DELETE etc.
-        if request.method != 'GET':
-            return HttpResponseForbidden()
-
-        # Add a message, and redirect to login.
-        if SEND_MESSAGE:
-            messages.info(request, _('You must be logged in to view this page.'))
-
-        return redirect_to_login(request.path_info)
+    """
+    login_exempt_urls = [settings.LOGIN_URL, settings.LOGOUT_URL]
+    login_exempt_urls += getattr(settings, 'LOGIN_EXEMPT_URLS', [])
+    login_protected_urls = getattr(settings, 'LOGIN_PROTECTED_URLS', [r'^'])
+
+    EXEMPT_URLS = compile_urls(login_exempt_urls)
+    PROTECTED_URLS = compile_urls(login_protected_urls)
+
+    SEND_MESSAGE = getattr(settings, 'LOGIN_REQUIRED_SEND_MESSAGE', True)
+
+    def __init__(self, check=True):
+        if check:
+            check_request_has_user()
+
+    def get_exempt_url_patterns(self):
+        return self.EXEMPT_URLS
+
+    def get_protected_url_patterns(self):
+        return self.PROTECTED_URLS
+
+    def get_access_denied_message(self):
+        if not self.SEND_MESSAGE:
+            return ''
+
+        return LoginPermissionMiddlewareMixin.get_access_denied_message(self)
```

Comment:
[Python 2 is slightly awkward.](http://stackoverflow.com/a/10759155)

Basically, to use `super`, I need to make `BasePermissionMiddleware` inherit from `object`.  That doesn't hurt, but since I was fixing it anyway, I preferred this as it's safer and a bit more explicit.

That said, I should make `BasePermissionMiddleware` inherit from `object` anyway so that people using the library can call `super`.